spec/models/distribution_spec.rb

```diff
@@ -169,6 +170,52 @@
       distribution = create(:distribution, created_at: yesterday)
       expect(distribution.issued_at).to eq(distribution.created_at)
     end
+
+    context "#before_save" do
+      context "when creating distribution" do
+        context "when delivery_method is shipped and shipping cost is negative" do
+          let(:distribution) { create(:distribution, delivery_method: "shipped", shipping_cost: -12.05) }
+
+          it "raises error" do
+            expect { distribution }.to raise_error(StandardError, "Shipping cost cannot be negative!")
+          end
+        end
+
+        context "when delivery_method is other then shipped" do
+          let(:distribution) { create(:distribution, delivery_method: "delivery", shipping_cost: 12.05) }
+
+          it "distribution will be created successfully and the shipping_cost will be zero" do
+            expect(distribution.errors).to be_empty
+            expect(distribution.shipping_cost).to eq(0)
+          end
+        end
+      end
+
+      context "when updating distribution" do
```

Comment:
There isn't really a need to test this twice. You want to use `valid?` here rather than create or update.